packages/@ember/-internals/glimmer/lib/component.ts

```diff
@@ -648,402 +649,431 @@ let lazyEventsProcessed = new WeakMap<EventDispatcher, WeakSet<object>>();
   @uses Ember.ViewStateSupport
   @public
 */
-const Component = CoreView.extend(
-  ChildViewsSupport,
-  ViewStateSupport,
-  ClassNamesSupport,
-  TargetActionSupport,
-  ActionSupport,
-  ViewMixin,
-  {
-    isComponent: true,
-
-    init() {
-      this._super(...arguments);
-      this[IS_DISPATCHING_ATTRS] = false;
-      this[DIRTY_TAG] = createTag();
-      this[BOUNDS] = null;
-
-      let eventDispatcher = this._dispatcher;
-      if (eventDispatcher) {
-        let lazyEventsProcessedForComponentClass = lazyEventsProcessed.get(eventDispatcher);
-        if (!lazyEventsProcessedForComponentClass) {
-          lazyEventsProcessedForComponentClass = new WeakSet<object>();
-          lazyEventsProcessed.set(eventDispatcher, lazyEventsProcessedForComponentClass);
-        }
+interface Component
+  extends CoreView,
+    ChildViewsSupport,
+    ViewStateSupport,
+    ClassNamesSupport,
+    TargetActionSupport,
+    ActionSupport,
+    ViewMixin {
+  parentView: Component | null;
+
+  /**
+    Layout can be used to wrap content in a component.
+    @property layout
+    @type Function
+    @public
+  */
+  layout?: TemplateFactory | Template;
+
+  /**
+    The name of the layout to lookup if no layout is provided.
+    By default `Component` will lookup a template with this name in
+    `Ember.TEMPLATES` (a shared global object).
+    @property layoutName
+    @type String
+    @default undefined
+    @private
+  */
+  layoutName?: string;
+
+  attributeBindings?: Array<string>;
+}
+class Component
```

Comment:
There is a (*small*) runtime risk here, if I remember correctly: there are some hazards around zebra-striping with classic classes. 🦓 CC @wycats @pzuraq @rwjblue @chancancode – I don’t remember what the issue was, but I do remember that we recommended against “zebra-striping” within chains, in the “it should work, 99% of the time, if you dot all your i’s and cross all your t’s but it’s easy to just not zebra-stripe” bucket. Does anyone remember the details, and if so is it an ongoing concern such that we *shouldn’t* do this conversion?

Reply:
To elaborate: the problem isn't internal, but for external consumers, since [we currently still document](https://api.emberjs.com/ember/4.1/classes/Component) `Component.extend({ ... })` as the preferred way to define a Classic component.

Reply:
The issue is that some properties passed in extend won't override properties on the base class. I'll investigate these cases.